tests/qat/test_quantum_backend.py

```diff
@@ -716,3 +716,23 @@ def test_create_duration_timeline_mapping(self):
             PhaseReset,
             *[PostProcessing] * 4,
         ]
+
+    def test_mid_circuit_validation(self):
+        """Tests that an error is thrown for mid-circuit measurements."""
+
+        hw = get_test_model()
+        engine = get_test_execution_engine(hw)
+        q0 = hw.get_qubit(0)
+        q1 = hw.get_qubit(1)
+        builder = (
+            get_builder(hw)
+            .had(q0)
+            .X(q1)
+            .measure(q0)
+            .measure(q1)
+            .cnot(q0, q1)
```

Comment:
Super minor but cloud we also add tests to validate:
1. measure only one of the qubits in the cnot before
2. measure e.g. q2 and then do gates on un-measured qubits

Reply:
Thanks, new tests coming right up! :smile: